core/src/main/java/org/dozer/MappingProcessor.java

```diff
@@ -517,7 +517,21 @@ private Object mapOrRecurseObject(Object srcObj, Object srcFieldValue, Class<?>
 
   private <T extends Enum<T>> T mapEnum(Enum<T> srcFieldValue, Class<T> destFieldType) {
     String name = srcFieldValue.name();
-    return Enum.valueOf(destFieldType, name);
+    return Enum.valueOf(getSerializableEnumClass(destFieldType), name);
+  }
+  
+  private <T extends Enum<T>> Class<T> getSerializableEnumClass(Class<T> enumClass) {
+    if (checkIfOverridenEnum(enumClass)) {
+      /* This cast is case because of the check performed by the checkIfOverridenEnum function. */
+      @SuppressWarnings("unchecked")
+      Class<T> castedSuperclass = (Class<T>) enumClass.getSuperclass();
+      return castedSuperclass;
+    }
+    return enumClass;
+  }
+
+  private <T extends Enum<T>> boolean checkIfOverridenEnum(Class<T> enumClass) {
+    return Enum.class.isAssignableFrom(enumClass.getSuperclass()) && !Enum.class.equals(enumClass.getSuperclass());
```

Comment:
`Enum.class.isAssignableFrom(enumClass.getSuperclass())` is redundant. It is already guaranteed that this code will be executed only for enums (see`MappingUtils.isEnumType` before `mapEnum` call).